src/providers/pkcs11/mod.rs

```diff
@@ -190,6 +178,26 @@ impl Provider {
 
         Some(pkcs11_provider)
     }
+
+    // Create a new session with the following properties:
+    // * without callback
+    // * read/write session
+    // * serial session
+    // * logged in if the pin is set
+    // * set on the slot in the provider
+    fn new_session(&self) -> Result<Session> {
+        let mut flags = Flags::new();
+        let _ = flags.set_rw_session(true).set_serial_session(true);
```

Comment:
A bit weird, but I guess it's ok (saying this because the setters for `Flags` are in the middle between normal setters and builder setters, so you can do this chaining, but the variable has to be mutable and you have to `let _` the last return).

Reply:
Yes, might be worth a rework at some point!